base/src/main/java/proguard/classfile/util/inject/CodeInjector.java

```diff
@@ -0,0 +1,216 @@
+
+  /**
+   * Specify the method where a static method invocation shall be injected into.
+   *
+   * @param programClass The program class that has the method where a static method invocation
+   *     shall be injected into.
+   * @param programMethod the method where a static method invocation shall be injected into.
+   */
+  public CodeInjector into(ProgramClass programClass, ProgramMethod programMethod) {
+    assert targets == null : "The injection target has already been specified.";
+
+    targets = Arrays.asList(new ClassMethodPair(programClass, programMethod));
+    return this;
+  }
+
+  /**
+   * Specify the location in which the invoke instruction should be injected into.
+   *
+   * @param injectStrategy The implementation of InjectStrategy interface which determine the offset
+   *     to inject the invoke instruction.
+   * @return
+   */
+  public CodeInjector at(InjectStrategy injectStrategy) {
+    assert this.injectStrategy == null
+        : "The injection strategy: " + injectStrategy + " has already been specified.";
+
+    this.injectStrategy = injectStrategy;
+    return this;
+  }
+
+  /**
+   * Apply the invoke instruction in accordance to the specifications provided via the
+   * `.injectInvokeStatic(...)`, `.into(...)` and `at(...)` method.
+   */
+  public void commit() {
+    assert content != null
+        : "The injection content hasn't been provided; please use `.injectInvokeStatic(...)` "
+            + "to indicate the method invocation to be injected.";
+
+    assert targets != null
+        : "The injection target hasn't been provided; please use `.into(...)` to indicate the method targeted for "
+            + "injecting "
+            + renderInjectionContent(content.clazz, content.method, arguments)
+            + ".";
+
+    assert injectStrategy != null
+        : "The injection location hasn't been provided. please use `.at(...)` to indicate the place to inject "
+            + renderInjectionContent(content.clazz, content.method, arguments)
+            + " into the target method.";
+
+    targets.forEach(
+        target -> {
+          CodeAttributeEditor editor = new CodeAttributeEditor();
+          InstructionSequenceBuilder code =
+              new InstructionSequenceBuilder((ProgramClass) target.clazz);
+
+          arguments.forEach(
+              argument ->
+                  code.pushPrimitiveOrString(argument.getValue(), argument.getInternalType()));
+          code.invokestatic(content.clazz, content.method);
+
+          target.method.accept(
+              target.clazz,
+              new AllAttributeVisitor(
+                  new AttributeNameFilter(
+                      Attribute.CODE,
+                      new AttributeVisitor() {
+                        @Override
+                        public void visitCodeAttribute(
+                            Clazz clazz, Method method, CodeAttribute codeAttribute) {
+                          editor.reset(codeAttribute.u4codeLength);
```

Comment:
this is more of a style preference, so consider this as a nit: I prefer his to be an explicit class even if it's only used here